tests/integration/pure_cairo_vm/cheatcodes/call/call_test.cairo

```diff
@@ -52,3 +52,23 @@ func test_call_named_args_invalid_fail() {
     %{ call(ids.deployed_contract_address, "increase_balance", {"xxx": 50}) %}
     return ();
 }
+
+func test_call_with_proxy(){
+    alloc_locals;
+
+    %{
+        target_addr = deploy_contract("./src/basic.cairo").contract_address
+        proxy_addr = deploy_contract("./src/proxy.cairo").contract_address
+
+        invoke(proxy_addr, "set_target", [target_addr])
+
+        assert call(proxy_addr, "get_balance") == [100]
+        call(proxy_addr, "increase_twice", [50])
+        call(target_addr, "increase_balance", [50])
+        assert call(proxy_addr, "get_balance") == [100]
+    %}
```

Comment:
I just find such a code easier to read/write. If you insist, I can expand this to a code like this:
```
    local deployed_contract_address;
    %{ ids.deployed_contract_address = deploy_contract("./src/basic.cairo").contract_address %}
    assert_not_zero(deployed_contract_address);
```